files/fr/web/javascript/enumerability_and_ownership_of_properties/index.md

```diff
@@ -0,0 +1,495 @@
+    },
+    _enumerableAndNotEnumerable : function (obj, prop) {
+        return true;
+    },
+    // Inspirée par https://stackoverflow.com/a/8024294/271577
+    _getPropertyNames : function getAllPropertyNames(obj, iterateSelfBool, iteratePrototypeBool, includePropCb) {
+        var props = [];
+
+        do {
+            if (iterateSelfBool) {
+                Object.getOwnPropertyNames(obj).forEach(function (prop) {
+                    if (props.indexOf(prop) === -1 && includePropCb(obj, prop)) {
+                        props.push(prop);
+                    }
+                });
+            }
+            if (!iteratePrototypeBool) {
+                break;
+            }
+            iterateSelfBool = true;
+        } while (obj = Object.getPrototypeOf(obj));
+
+        return props;
+    }
+};
+```
+
+## Tableau de détection
+
+<table>
+  <thead>
+    <tr>
+      <th scope="row"></th>
+      <th scope="col"><code>in</code></th>
+      <th scope="col"><code>for..in</code></th>
+      <th scope="col"><code>obj.hasOwnProperty()</code></th>
+      <th scope="col"><code>obj.propertyIsEnumerable()</code></th>
+      <th scope="col"><code>Object.keys()</code></th>
+      <th scope="col"><code>Object.getOwnPropertyNames()</code></th>
+      <th scope="col"><code>Object.getOwnPropertyDescriptors()</code></th>
+      <th scope="col"><code>Reflect.ownKeys()</code></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th scope="row">Propriétés énumérables</th>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+    </tr>
+    <tr>
+      <th scope="row">Propriétés non-énumérables</th>
+      <td>true</td>
+      <td>false</td>
+      <td>true</td>
+      <td>false</td>
+      <td>false</td>
+      <td>true</td>
+      <td>true</td>
+      <td>true</td>
+    </tr>
+    <tr>
+      <th scope="row">Propriétés dont les clés sont des symboles</th>
+      <td>true</td>
+      <td>false</td>
+      <td>true</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>true</td>
+      <td>true</td>
+    </tr>
+    <tr>
+      <th scope="row">Propriétés héritées et énumérables</th>
+      <td>true</td>
+      <td>true</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+    </tr>
+    <tr>
+      <th scope="row">Propriétés héritées et non-énumérables</th>
+      <td>true</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+    </tr>
+    <tr>
+      <th scope="row">Propriétés héritées dont les clés sont des symboles</th>
+      <td>true</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+      <td>false</td>
+    </tr>
+  </tbody>
+</table>
```

Comment:
Tableau non converti

Reply:
@tristantheb https://raw.githubusercontent.com/mdn/content/main/files/en-us/web/javascript/enumerability_and_ownership_of_properties/index.md "works for me" I guess ^ - ^